ISISNeutronMuon/InstrumentScripts
# Background Plot: Handle errors when loading from save file

### Instrument(s)

Technique, Muon

### Story/Acceptance criteria

The background plot task does not fall over if it's save files are corrupted

### Description of work

Handle invalid data point when appending to dataset, set `run_numbers_match` to False on invalid `saved_run_number`/`current_run_number`, handle row index & datetime format errors.
Display corrupt file warnings and attempt to load and plot using the valid data from the file.

### Issue/Ticket Reference

https://github.com/ISISComputingGroup/IBEX/issues/5582

### Tests

Manual tests. 

### To test

[Background Script IOC](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Background-Script-IOC)

Configuration, `background_script.py`, `init_<name>.py`, IOC logs and "corrupted" save file used for testing:
[Ticket5582.zip](https://github.com/ISISNeutronMuon/InstrumentScripts/files/5822006/Ticket5582.zip)

Run the background plot as normal via the `BGRSCRPT_01` IOC, check that the plot is working as intended.
Edit the `background_plot_data_01.csv` file to corrupt the data.
Restart IOC, make sure the dataset matches (`self.saved_data_matches_current_dataset()`) in `BackgroundPlot.start` so the script loads the save file instead of creating a new file.
Check that the IOC does not get stuck in a restart loop due to failing to load the save file.

---

#### Code Review


### Final Steps
